CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/gui/BooleanInputPane.java

```diff
@@ -95,6 +96,12 @@ public BooleanInputPane(final PluginParameter<BooleanParameterValue> parameter)
             });
         });
 
-        getChildren().add(field);
+        parameterId = parameter.getId();
+        List<String> booleanInputRecentValues = RecentParameterValues.getRecentValues(parameterId);
+        if (booleanInputRecentValues.size() > 1) {
+            parameter.setStringValue(booleanInputRecentValues.get(1));
+        } else {
+            parameter.setStringValue(booleanInputRecentValues.get(0));
+        }
```

Comment:
Firstly what is the reason when the size is greater than 1 you have to use index 1?

Anyway an optimisation could be as follows though it may be difficult to read:
```
parameter.setStringValue(booleanInputRecentValues.get(booleanInputRecentValues.size() > 1? 1 : 0));
```

Other *InputPane classes have the same pattern going on so perhaps they could all be changed or perhaps a new utility class could be used to hold this logic?